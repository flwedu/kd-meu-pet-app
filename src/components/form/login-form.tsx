import { FormEvent, useState } from "react";

export function LoginForm() {
  const [data, setData] = useState({ username: "", password: "" });

  function handleChange(e: FormEvent<HTMLInputElement>) {
    const { name, value } = e.currentTarget;

    setData({ ...data, [name]: value });
  }

  function handleSubmit() {
    console.log("Submitted");
  }

  return (
    <form
      className="form form-login align-center flex flex-col rounded"
      onSubmit={handleSubmit}
    >
      <div onChange={handleChange}>
        <h2>Bem vindo!</h2>
        <p>Por favor, se identifique:</p>
        <div className="field">
          <input
            type="text"
            name="username"
            className="input"
            placeholder="UsuÃ¡rio / E-mail"
            required
          />
        </div>
        <div className="field">
          <input
            type="password"
            name="password"
            className="input"
            placeholder="Senha"
            required
          />
        </div>
      </div>
    </form>
  );
}
