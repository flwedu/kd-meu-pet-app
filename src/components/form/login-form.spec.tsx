import { fireEvent, render } from "@testing-library/react";
import { LoginForm } from "./login-form";

describe("LoginForm component tests", () => {
  const onSubmit = jest.fn();
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("should render the component properly with the username and password fields", () => {
    const { container } = render(<LoginForm onSubmit={onSubmit} />);

    expect(container).toMatchSnapshot();
  });

  test("should call onSubmit only if the username and password fields are valid", () => {
    const { getByPlaceholderText } = render(<LoginForm onSubmit={onSubmit} />);

    const usernameInput = getByPlaceholderText(
      "Nome de usu√°rio"
    ) as HTMLInputElement;
    const passwordInput = getByPlaceholderText("Senha") as HTMLInputElement;

    const form = usernameInput.form!;
    fireEvent.submit(form);

    expect(onSubmit).not.toHaveBeenCalled();

    fireEvent.change(usernameInput, { target: { value: "test" } });
    fireEvent.change(passwordInput, { target: { value: "test" } });

    fireEvent.submit(form);

    expect(onSubmit).toHaveBeenCalledTimes(1);
  });
});

export {};
