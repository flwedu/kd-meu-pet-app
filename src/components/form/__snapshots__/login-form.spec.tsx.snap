// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component tests should render the component properly 1`] = `
<div>
  <form
    class="form form-login align-center flex flex-col rounded"
  >
    <div>
      <h2>
        Bem vindo!
      </h2>
      <p>
        Por favor, se identifique:
      </p>
      <div
        class="field"
      >
        <input
          class="input"
          name="username"
          placeholder="UsuÃ¡rio / E-mail"
          required=""
          type="text"
        />
      </div>
      <div
        class="field"
      >
        <input
          class="input"
          name="password"
          placeholder="Senha"
          required=""
          type="password"
        />
      </div>
    </div>
  </form>
</div>
`;
